tests/Endpoint/PartitionEndpointProviderTest.php

```diff
@@ -0,0 +1,184 @@
+            ],
+            [
+                ['region' => 'us-gov-west-1', 'service' => 'iam'],
+                [
+                    'endpoint' => 'https://iam.us-gov.amazonaws.com',
+                    'signatureVersion' => 'v4',
+                    'signingRegion' => 'us-gov-west-1',
+                    'signingName' => 'iam',
+                ],
+            ],
+            [
+                ['region' => 'us-gov-west-1', 'service' => 's3'],
+                [
+                    'endpoint' => 'https://s3-us-gov-west-1.amazonaws.com',
+                    'signatureVersion' => 's3v4',
+                    'signingRegion' => 'us-gov-west-1',
+                    'signingName' => 's3',
+                ],
+            ],
+            [
+                ['region' => 'us-gov-baz', 'service' => 'foo'],
+                [
+                    'endpoint' => 'https://foo.us-gov-baz.amazonaws.com',
+                    'signatureVersion' => 'v4',
+                    'signingRegion' => 'us-gov-baz',
+                    'signingName' => 'foo',
+                ],
+            ],
+            [
+                ['region' => 'cn-north-1', 'service' => 's3'],
+                [
+                    'endpoint' => 'https://s3.cn-north-1.amazonaws.com.cn',
+                    'signatureVersion' => 's3v4',
+                    'signingRegion' => 'cn-north-1',
+                    'signingName' => 's3',
+                ],
+            ],
+            [
+                ['region' => 'cn-north-1', 'service' => 'ec2'],
+                [
+                    'endpoint' => 'https://ec2.cn-north-1.amazonaws.com.cn',
+                    'signatureVersion' => 'v4',
+                    'signingRegion' => 'cn-north-1',
+                    'signingName' => 'ec2',
+                ],
+            ],
+            [
+                ['region' => 'local', 'service' => 'dynamodb', 'scheme' => 'http'],
+                [
+                    'endpoint' => 'http://localhost:8000',
+                    'signatureVersion' => 'v4',
+                    'signingRegion' => 'us-east-1',
+                    'signingName' => 'dynamodb',
+                ],
+            ],
+            [
+                ['region' => 'us-east-1', 'service' => 'iot'],
+                [
+                    'endpoint' => 'https://iot.us-east-1.amazonaws.com',
+                    'signatureVersion' => 'v4',
+                    'signingRegion' => 'us-east-1',
+                    'signingName' => 'execute-api',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider partitionRegionProvider
+     *
+     * @param string $region
+     * @param string $partition
+     */
+    public function testResolvesPartitionsByRegion($region, $partition)
+    {
+        // Use the default endpoints file
+        $p = PartitionEndpointProvider::defaultProvider();
+        $this->assertSame($partition, $p->getPartitionFromRegion($region)['partition']);
+    }
```

Comment:
In consideration for completeness and consistency, how about adding the test for unknown region and check whether default region is added as well?